presto-iceberg/src/main/java/com/facebook/presto/iceberg/util/StatisticsUtil.java

```diff
@@ -57,29 +68,49 @@ public static TableStatistics mergeHiveStatistics(TableStatistics icebergStatist
                     .setDistinctValuesCount(icebergColumnStats.getDistinctValuesCount())
                     .setRange(icebergColumnStats.getRange());
             if (hiveColumnStats != null) {
-                if (mergeStrategy.equals(USE_NDV) || mergeStrategy.equals(USE_NULLS_FRACTION_AND_NDV)) {
+                // NDVs
+                if (mergeFlags.contains(NUMBER_OF_DISTINCT_VALUES)) {
                     hiveColumnStats.getDistinctValuesCount().ifPresent(ndvs -> mergedStats.setDistinctValuesCount(Estimate.of(ndvs)));
                 }
-                if (mergeStrategy.equals(USE_NULLS_FRACTIONS) || mergeStrategy.equals(USE_NULLS_FRACTION_AND_NDV)) {
-                    hiveColumnStats.getNullsCount().ifPresent(nullCount -> {
-                        Estimate nullsFraction;
-                        if (!hiveStatistics.getBasicStatistics().getRowCount().isPresent()) {
-                            if (icebergStatistics.getRowCount().isUnknown()) {
-                                nullsFraction = Estimate.unknown();
-                            }
-                            else {
-                                nullsFraction = Estimate.of((double) nullCount / icebergStatistics.getRowCount().getValue());
-                            }
-                        }
-                        else {
-                            nullsFraction = Estimate.of((double) nullCount / hiveStatistics.getBasicStatistics().getRowCount().getAsLong());
-                        }
-                        mergedStats.setNullsFraction(nullsFraction);
-                    });
+                // data size
+                if (mergeFlags.contains(ColumnStatisticType.TOTAL_SIZE_IN_BYTES)) {
+                    hiveColumnStats.getTotalSizeInBytes().ifPresent(size -> mergedStats.setDataSize(Estimate.of(size)));
                 }
             }
             statsBuilder.setColumnStatistics(columnHandle, mergedStats.build());
         });
-        return statsBuilder.build();
+        return calculateAndSetTableSize(statsBuilder).build();
+    }
+
+    public static EnumSet<ColumnStatisticType> decodeMergeFlags(String input)
+    {
+        return Optional.of(Arrays.stream((input).trim().split(","))
+                        .filter(value -> !value.isEmpty())
+                        .map(ColumnStatisticType::valueOf)
+                        .collect(Collectors.toSet()))
+                .filter(set -> !set.isEmpty())
+                .map(EnumSet::copyOf)
+                .orElse(EnumSet.noneOf(ColumnStatisticType.class));
+    }
+
+    public static String encodeMergeFlags(EnumSet<ColumnStatisticType> flags)
+    {
+        return Joiner.on(",").join(flags.stream().map(Enum::name).iterator());
+    }
+
+    public static TableStatistics.Builder calculateAndSetTableSize(TableStatistics.Builder builder)
```

Comment:
Let's move this static utility to the TableStatistics.Builder class itself ?

Reply:
I can't because it relies on casting to `IcebergColumnHandle` to check if the type is a `FixedWidthType`. The SPI for `ColumnHandle` has no methods which would allow this